[bug] crc setup fail on windows as `crc daemon` hits panic to invalid pointer reference

crc version: 05.20 main branch
bundle version: 4.18.2
platform: windows

```
PS C:\Users\crcqe> crc setup
CRC is constantly improving and we would like to know more about usage (more details at https://developers.redhat.com/article/tool-data-collection)
Your preference can be changed manually if desired using 'crc config set consent-telemetry <yes/no>'
Would you like to contribute anonymous usage statistics? [y/N]: y
Thanks for helping us! You can disable telemetry with the command 'crc config set consent-telemetry no'.
INFO Using bundle path C:\Users\crcqe\.crc\cache\crc_hyperv_4.18.2_amd64.crcbundle
INFO Checking minimum RAM requirements
INFO Check if Podman binary exists in: C:\Users\crcqe\.crc\bin\oc
INFO Checking if running in a shell with administrator rights
INFO Checking Windows release
INFO Checking Windows edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if Hyper-V service is enabled
INFO Checking if crc-users group exists
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Checking if vsock is correctly configured
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if C:\Users\crcqe\.crc\cache\crc_hyperv_4.18.2_amd64.crcbundle exists
INFO Checking if the win32 background launcher is installed
INFO Checking if the daemon task is installed
INFO Installing the daemon task
INFO Checking if the daemon task is running
INFO Running the daemon task
Temporary error: daemon is not running yet (x8)
```

---

running `crc daemon --log-level debug` on windows is failing with following error:

```
PS C:\Users\rhqp> crc daemon --log-level debug
DEBU CRC version: 2.50.0+8caa80
DEBU OpenShift version: 4.18.2
DEBU MicroShift version: 4.18.2
DEBU Running 'crc daemon'
INFO listening \\.\pipe\crc-http
INFO listening vsock://00000400-FACB-11E6-BD58-64006A7986D3
0B sent to the VM, 0B received from the VM
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x7ff6934791d6]

goroutine 92 [running]:
github.com/crc-org/crc/v2/cmd/crc/cmd.run.func4()
        D:/a/crc/crc/cmd/crc/cmd/daemon.go:221 +0x56
created by github.com/crc-org/crc/v2/cmd/crc/cmd.run in goroutine 1
        D:/a/crc/crc/cmd/crc/cmd/daemon.go:212 +0x345

```

--- a/cmd/crc/cmd/daemon.go
+++ b/cmd/crc/cmd/daemon.go
@@ -14,17 +14,17 @@ import (
 	"syscall"
 	"time"
 
-	"github.com/crc-org/crc/v2/pkg/crc/api/client"
-	"github.com/crc-org/crc/v2/pkg/crc/daemonclient"
-
 	"github.com/containers/gvisor-tap-vsock/pkg/types"
 	"github.com/containers/gvisor-tap-vsock/pkg/virtualnetwork"
 	"github.com/crc-org/crc/v2/pkg/crc/adminhelper"
 	"github.com/crc-org/crc/v2/pkg/crc/api"
+	"github.com/crc-org/crc/v2/pkg/crc/api/client"
 	"github.com/crc-org/crc/v2/pkg/crc/api/events"
 	crcConfig "github.com/crc-org/crc/v2/pkg/crc/config"
 	"github.com/crc-org/crc/v2/pkg/crc/constants"
+	"github.com/crc-org/crc/v2/pkg/crc/daemonclient"
 	"github.com/crc-org/crc/v2/pkg/crc/logging"
+	"github.com/crc-org/machine/libmachine/drivers"
 	"github.com/docker/go-units"
 	"github.com/gorilla/handlers"
 	"github.com/pkg/errors"
@@ -214,11 +214,16 @@ func run(configuration *types.Configuration) error {
 		for {
 			ctx, cancel := context.WithCancel(context.Background())
 			conn, err := unixgramListener(ctx, vn)
+			if err != nil && errors.Is(err, drivers.ErrNotImplemented) {
+				cancel()
+				break
+			}
 			if err != nil && !errors.Is(err, net.ErrClosed) {
 				logging.Errorf("unixgramListener error: %v", err)
 			}
+
 			if oldCancel != nil {
-				logging.Warnf("New connection from %s. Closing old connection", conn.LocalAddr().String())
+				logging.Warnf("New connection to %s. Closing old connection", conn.LocalAddr().String())
 				oldCancel()
 			}
 			oldCancel = cancel

--- a/cmd/crc/cmd/daemon_darwin.go
+++ b/cmd/crc/cmd/daemon_darwin.go
@@ -39,7 +39,7 @@ func unixgramListener(ctx context.Context, vn *virtualnetwork.VirtualNetwork) (*
 	if err != nil {
 		return conn, errors.Wrap(err, "failed to listen unixgram")
 	}
-	logging.Infof("listening on %s:", constants.UnixgramSocketPath)
+	logging.Infof("listening on %s", constants.UnixgramSocketPath)
 	vfkitConn, err := transport.AcceptVfkit(conn)
 	if err != nil {
 		return conn, errors.Wrap(err, "failed to accept vfkit connection")

--- a/cmd/crc/cmd/daemon_linux.go
+++ b/cmd/crc/cmd/daemon_linux.go
@@ -8,11 +8,11 @@ import (
 
 	"github.com/containers/gvisor-tap-vsock/pkg/transport"
 	"github.com/containers/gvisor-tap-vsock/pkg/virtualnetwork"
-	"github.com/crc-org/crc/v2/pkg/crc/constants"
-	"github.com/crc-org/crc/v2/pkg/crc/logging"
-
 	"github.com/coreos/go-systemd/v22/activation"
 	"github.com/coreos/go-systemd/v22/daemon"
+	"github.com/crc-org/crc/v2/pkg/crc/constants"
+	"github.com/crc-org/crc/v2/pkg/crc/logging"
+	"github.com/crc-org/machine/libmachine/drivers"
 	"github.com/mdlayher/vsock"
 )
 
@@ -127,7 +127,7 @@ func httpListener() (net.Listener, error) {
 }
 
 func unixgramListener(_ context.Context, _ *virtualnetwork.VirtualNetwork) (*net.UnixConn, error) {
-	return nil, nil
+	return nil, drivers.ErrNotImplemented
 }
 
 func startupDone() {

--- a/cmd/crc/cmd/daemon_windows.go
+++ b/cmd/crc/cmd/daemon_windows.go
@@ -9,6 +9,7 @@ import (
 	"github.com/containers/gvisor-tap-vsock/pkg/virtualnetwork"
 	"github.com/crc-org/crc/v2/pkg/crc/constants"
 	"github.com/crc-org/crc/v2/pkg/crc/logging"
+	"github.com/crc-org/machine/libmachine/drivers"
 )
 
 func vsockListener() (net.Listener, error) {
@@ -38,7 +39,7 @@ func checkIfDaemonIsRunning() (bool, error) {
 }
 
 func unixgramListener(_ context.Context, _ *virtualnetwork.VirtualNetwork) (*net.UnixConn, error) {
-	return nil, nil
+	return nil, drivers.ErrNotImplemented
 }
 
 func startupDone() {

